pkg/utils/k8sutil.go

```diff
@@ -969,3 +970,11 @@ func InitializeEmptyMapsInDeployment(deployment *v1beta2.Deployment) {
 		deployment.Spec.Template.Spec.NodeSelector = make(map[string]string)
 	}
 }
+
+// MergeDeployments merges two deployment objects
+func MergeDeployments(destinationDeployment *v1beta2.Deployment, sourceDeployment *v1beta2.Deployment) error {
```

Comment:
nitpicking comment: we should either not modify the content of the deployment pointers or document that this function modifies them

Reply:
Np problem :) 
Here function signature has a pointer, isn't that self-explanatory and makes it readable? IMO such a comment is redundant. Ofcourse, `pass by value` is another way of doing it.

Reply:
yes, it is almost explicit but it could be the case that you are using the reference because it doesn't create a copy of the object in memory and it is slightly more efficient. In theory it is not necessary to modify the objects to merge them, that's why it may be confusing. Anyway, it is not that important.